src/library-authoring/components/messages.ts

```diff
@@ -156,5 +156,30 @@ const messages = defineMessages({
     defaultMessage: 'Unpublished changes',
     description: 'Badge text shown when a component has unpublished changes',
   },
+  publishConfirmationTitle: {
+    id: 'course-authoring.library-authoring.component.publish-confirmation.title',
+    defaultMessage: 'Publish {displayName}',
+    description: 'Title of the modal to confirm publish a component in a library',
+  },
+  publishConfirmationButton: {
+    id: 'course-authoring.library-authoring.component.publish-confirmation.confirm',
+    defaultMessage: 'Publish',
+    description: 'Text in confirmation button of the modal to confirm publish a component in a library',
+  },
+  publishConfirmationBody: {
+    id: 'course-authoring.library-authoring.component.publish-confirmation.body',
+    defaultMessage: 'Publish all unpublished changes for this component?',
+    description: 'Body text of the modal to confirm publish a component in a library',
+  },
+  publishConfimrationDownstreamsBody: {
+    id: 'course-authoring.library-authoring.component.publish-confirmation.downsteams-body',
+    defaultMessage: 'This content is currently being used in:',
+    description: 'Body text to show downstreams of the modal to confirm publish a component in a library',
+  },
+  publishConfirmationDownstreamsAlert: {
+    id: 'course-authoring.library-authoring.component.publish-confirmation.downsteams-alert',
+    defaultMessage: 'Publishing this component will push the changes out to the courses.',
```

Comment:
This is not the actual behaviour of our publish: changing a component will not automatically push the changes downstream.

Could you confirm which behaviour is desired here?

Reply:
I asked to product: https://github.com/openedx/frontend-app-authoring/issues/1188#issuecomment-2691629603

Reply:
Updated: https://github.com/openedx/frontend-app-authoring/pull/1677/commits/3ca0813ba2ea3719f86875ccae20be1570dab066